src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -465,6 +464,71 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
             }
         }
 
+        private static KeyValuePair<string, string> GenerateCookie(string name, char repeat, int overallHeaderValueLength)
+        {
+            string emptyHeaderValue = $"{name}=; Path=/";
+
+            Debug.Assert(overallHeaderValueLength > emptyHeaderValue.Length);
+
+            int valueCount = overallHeaderValueLength - emptyHeaderValue.Length;
+            string value = new string(repeat, valueCount);
+
+            return new KeyValuePair<string, string>(name, value);
+        }
+
+        public static object[][] CookieNameValues =
+        {
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 126) },
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 127) },
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 128) },
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 129) },
+
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 254) },
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 255) },
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 256) },
+            new object[] { GenerateCookie(name: "foo", repeat: 'a', overallHeaderValueLength: 257) },
```

Comment:
Does HTTP define a max cookie size?  Considering adding one for that if so.  I see that CookieContainer defines a default max size... maybe ones around that.

Also, it'd be good to include a comment about why these values were chosen, so that someone reading the code in a few years understands why they're there.